cl/corpus_importer/management/commands/update_casenames_wl_dataset.py

```diff
@@ -0,0 +1,360 @@
+    docket_case_name_updated = False
+
+    if not matched_cluster.case_name:
+        # Save case name in cluster when we don't have it
+        matched_cluster.case_name = csv_case_name
+        matched_cluster.save()
+        logger.info(f"Case name updated for cluster id: {matched_cluster.id}")
+        cluster_case_name_updated = True
+
+    if not matched_cluster.docket.case_name:
+        # Save case name in docket when we don't have it
+        matched_cluster.docket.case_name = csv_case_name
+        matched_cluster.docket.save()
+        logger.info(
+            f"Case name updated for docket id: {matched_cluster.docket.id}"
+        )
+        docket_case_name_updated = True
+
+    return cluster_case_name_updated, docket_case_name_updated
+
+
+def process_csv(
+    filepath: str, delay: float, dry_run: bool, chunk_size: int
+) -> None:
+    """Process rows from csv file
+
+    :param filepath: path to csv file
+    :param delay: delay between saves in seconds
+    :param dry_run: flag to simulate update process
+    :param chunk_size: number of rows to read at a time
+    """
+
+    total_clusters_updated = 0
+    total_dockets_updated = 0
+
+    logger.info(f"Processing {filepath}")
+    for chunk in pd.read_csv(filepath, chunksize=chunk_size):
+        for row in chunk.dropna().itertuples():
+            (
+                index,
+                west_case_name,
+                court,
+                date_str,
+                cite1,
+                cite2,
+                docket,
+                volume,
+            ) = row
+
+            clean_docket_num = docket.strip('="').strip('"')
+            if not clean_docket_num:
+                logger.info(f"Row index: {index} - No docket number found.")
+                continue
+
+            date_filed = parse_date(date_str)
+            if not date_filed:
+                logger.info(f"Row index: {index} - No valid date found.")
+                continue
+
+            valid_citations = parse_citations([cite1, cite2])
+
+            if not valid_citations:
+                logger.info(f"Row index: {index} - Missing two citations.")
+                continue
+
+            # Query for possible matches using data from row
+            possible_matches = query_possible_matches(
+                valid_citations=valid_citations,
+                docket_number=clean_docket_num,
+                date_filed=date_filed,
+            )
+
+            if not possible_matches:
+                logger.info(f"Row index: {index} - No matches found.")
+                continue
+
+            matches = []
+            for match in possible_matches:
+                cl_case_name = (
+                    match.cluster.case_name_full
+                    if match.cluster.case_name_full
+                    else match.cluster.case_name
+                )
+                case_name_match = check_case_names_match(
+                    west_case_name, cl_case_name
+                )
```

Comment:
The regex needs some tweaking, it almost works fine, i found some cases where there are spaces between abbreviations:

`"In re P. I.","Court of Appeal, First District, Division 3, California.","January 20, 1989","207 Cal.App.3d 316","254 Cal.Rptr. 774","A041221","254"` https://www.courtlistener.com/opinion/2168480/in-re-pi/

`"In re A. M.","Court of Appeal, First District, Division 2, California.","December 01, 1989","216 Cal.App.3d 319","264 Cal.Rptr. 666","A042237","264"`  https://www.courtlistener.com/opinion/2175113/in-re-am/

Reply:
what do you think about this:

```
def combine_initials(case_name: str) -> str:
    """Combine initials in case captions

    :param case_name: the case caption
    :return: the cleaned case caption
    """
    pattern = r"\b[A-Z](?:[A-Z\.]|\s)*[A-Z]\b\."

    return re.sub(
        pattern,
        lambda match: match.group(0).replace(" ", "").replace(".", ""),
        case_name,
    )
```